packages/zwave-js/src/lib/controller/Controller.ts

```diff
@@ -891,6 +904,53 @@ export class ZWaveController extends EventEmitter {
 		return true; // Don't invoke any more handlers
 	}
 
+	/**
+	 * Is called when an ReplaceFailed request is received from the controller.
+	 * Handles and controls the replace process.
+	 */
+	private async handleReplaceNodeRequest(
+		msg: ReplaceFailedNodeRequestStatusReport,
+	): Promise<boolean> {
+		log.controller.print(
+			`handling replace node request (status = ${
+				ReplaceFailedNodeStatus[msg.replaceStatus]
+			})`,
+		);
+
+		switch (msg.replaceStatus) {
+			case ReplaceFailedNodeStatus.NodeOK:
+				this._replaceFailedPromise?.reject(
+					new ZWaveError(
+						`The node could not be replaced because it's not failed`,
+						ZWaveErrorCodes.ReplaceFailedNode_NodeOK,
+					),
+				);
+				break;
+			case ReplaceFailedNodeStatus.FailedNodeReplaceFailed:
+				this._replaceFailedPromise?.reject(
+					new ZWaveError(
+						`The replace process could not be completed`,
+						ZWaveErrorCodes.ReplaceFailedNode_Failed,
+					),
+				);
+				break;
+			case ReplaceFailedNodeStatus.FailedNodeReplace:
+				// failed node is now ready to be replaced and controller is ready to add a new
+				// node with the nodeID of the failed node
+				await this.beginInclusion(this._includeNonSecure);
```

Comment:
I'm not sure this is correct, I think that the controller is automatically set in inclusion mode, the question is how do I specify if the inclusion should be done securely or not?

Reply:
> how do I specify if the inclusion should be done securely or not?

You already did by setting the `_includeNonSecure` variable. Remove the call to `beginInclusion` and set the `inclusionActive` variable so a user can't start a second inclusion process in parallel.

I'm thinking if the replaceFailed promise should be resolved here instead of after the replacement. At least that would be in line with `beginInclusion` and the library user would have some kind of notification that the new node can be included now.

Reply:
> I'm thinking if the replaceFailed promise should be resolved here instead of after the replacement. At least that would be in line with beginInclusion and the library user would have some kind of notification that the new node can be included now.

I think you right!

Reply:
But anyway the reported status seems wrong

Reply:
Operation: 

ReplaceFailedNode ---> reported status 3 (replace failed)
Put the device in inclusion mode --->  reported status 4 (unknown)

Restarting the controller results in the node to be added. I'm confused I think that the status report enum is wrong

Reply:
> But anyway the reported status seems wrong [...]
> Restarting the controller results in the node to be added

The stick has an internal state machine that easily gets confused, especially if you put it inclusion/exclusion mode multiple times. To be sure that your implementation is correct, I would pull the stick before every test.